autogen.sh

```diff
@@ -1,6 +1,29 @@
 #!/bin/sh
 
 rm -rf autom4te.cache
-mkdir -p config/m4 config/aux 
+mkdir -p config/m4 config/aux
+
+for arg in "$@"
+do
+    case $arg in
+        "--submodules")
+           echo "Updating git submodules..." && git submodule update --init --recursive && patch -fsp1 << EOF
```

Comment:
no need to call ```patch``` command - it will modify `configure.ac` which is under git control and file/repo will have conflicts.
The clean solution will be adding to `configure.ac` something like:
```
m4_ifdef([HAS_UCG],m4_include[src/ucg/configure.m4])
```

Reply:
@mike-dubman I'm not sure this example applies to the UCG case: in the example the m4_include macro applies to a file that always exists (e.g. m4/ax_cuda.m4) because it is part of the repository, whereas in UCG you propose "src/ucg/configure.m4" to be optionally pulled by autogen.sh (and not exist by default). 

Also, in your example `m4_ifdef` is similar to the preprocessor token `#ifndef` - which protects from double include: `m4_ifdef([macro], [if-defined], [if-not-defined])`. I think it's not exactly what we need. I can't see how `HAS_UCG` from your example could be controlled by _autogen.sh_ .

I agree patching configure.ac may be problematic (that's why I suggested _autogen.sh_ would pull unconditionally), but I don't see a way to condition `m4_include` from _autogen.sh_ without changing tracked files in the repo (and I'm still open for suggestions).

Reply:
m4_include should take `ucg.m4` from location attached by `git module fetch` from autogen.sh ? it does not have to be `m4/...` which resides in ucx, it can be from location provided by module.

`HAS_UCG` is set by configure if `./configure -with-ucg` is provided hence can be used later in the file to attach additional file.
what do i miss?

Reply:
@mike-dubman m4 macros run by _autoreconf_ in _autogen.sh_, before `configure` is even called. so using `--with-ucg` cannot control if `m4_include` is called or not.

Reply:
ohh.. i see what you mean, HAS_UCG is set by `ucx/configure.ac` when module already fetched.
but you can use `WANT_UCG` variable and set it in if `configure --enable-ucg` flag provided and then include ucg file if WANT_UCG is set

WDYT

Reply:
@mike-dubman  but I can't see a way to set WANT_UCG conditionally... it has to be set before `autoreconf` converts _configure.ac_ into _configure_. I don't see a way how _autogen.sh_ could set or unset WANT_UCG before calling `autoreconf`.

P.S. see whatsapp :)